src/main/java/com/example/demo/matcher/services/OrderService.java

```diff
@@ -1,23 +1,103 @@
 package com.example.demo.matcher.services;
 
 import com.example.demo.matcher.models.Order;
+import com.example.demo.matcher.models.OrderAction;
+import com.example.demo.matcher.models.OrderbookItem;
 
-import java.util.ArrayList;
+import java.util.*;
+import java.util.stream.Collectors;
 
 public class OrderService {
-    private ArrayList<Order> orders;
+    private final List<Order> orders;
 
-    public ArrayList<Order> get() {
+    public OrderService() {
+        orders = new ArrayList<>();
+    }
+
+    public List<Order> get() {
         return orders;
     }
 
     public void add(Order order) {
-
+        orders.add(order);
     }
 
     public void remove(Order order) {
+        orders.remove(order);
+    }
 
+    public List<OrderbookItem> getOrderbook(OrderAction action, String accountId) {
+        // filter the order list by action
+        List<Order> filtered = this.get().stream()
+                .filter(order -> order.getAction() == action && Objects.equals(order.getAccountId(), accountId))
+                .collect(Collectors.toCollection(ArrayList::new));
```

Comment:
Would this work here instead: ` .collect(Collectors.toList())`

Reply:
done